src/lib/support/Pool.h

```diff
@@ -166,6 +186,46 @@ struct HeapObjectList : HeapObjectListNode
 
 } // namespace internal
 
+/// Provides iteration over active objects in a Bitmap pool.
+///
+/// Creating and releasing items within a pool does not invalidate
+/// an iterator, however there are no guarantees which objects the
+/// iterator will return (i.e. newly created objects while iterating
+/// may be visible or not to the iterator depending where they are
+/// allocated).
+///
+/// You are not prevented from releasing the object the iterator
+/// currently points at. In that case, iterator should be advanced.
+template <class T>
+class BitmapActiveObjectIterator
```

Comment:
What about iterating over non-bitmap pool. Can we eventually scale to having "unbounded" pools,when allocated on heap?

Reply:
Heap pools have their own iterator: we already needed that because we needed `ForEachActiveObject`